mods/default/craftitems.lua

```diff
@@ -133,11 +133,14 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 			end
 		end
 
-		local data_str = minetest.serialize(data)
-		stack:set_metadata(data_str)
+		stack:get_meta():from_table(data)
 		book_on_use(stack, player)
 	end
 
+	-- Backwards compatibility
+	stack:set_metadata("")
```

Comment:
Rather than doing this, I decided to essentially return the modified stack from `book_on_use`. Yes, it was kinda fine as is and worked (and still does, BTW), however, it was inconsistent. If a future contributor were to modify the itemstack slightly more in `book_on_use`, they would be confused as to why it didn't update and yet this does.